src/pde_solve_ns.jl

```diff
@@ -102,5 +110,78 @@ function DiffEqBase.solve(
 
     Flux.train!(loss_n_sde, ps, data, opt; cb = cb)
 
-    save_everystep ? iters : re1(p3)(X0)[1]
+
+    if give_limit == false
+        save_everystep ? iters : re1(p3)(X0)[1]
+    else
```

Comment:
dt is required for the part, so check if `dt in keys(kwargs) || error("`dt` choice is required for lower bound calculation")`

Reply:
If I dont pass `dt` it throws this error : `Fixed timestep methods require a choice of dt or choosing the tstops` thats why I added the dt to this https://github.com/SciML/NeuralNetDiffEq.jl/blob/61bbef40706690515dc6022590b8139127f75a0e/src/pde_solve_ns.jl#L85

Reply:
oh yes, if you use a fixed timestep method it needs `dt`, but that's why the kwargs are splatted

Reply:
Oh, they were there earlier, but even before in all the test cases `dt` was passed which I found strange so I added `dt` as a direct argument..

Reply:
The test gave the same error. Should I restore the `dt` ?

Reply:
Make `dt` default to `eltype(concrete_prob(_prob).tspan)(0)` and then pass it along, and when you use it explicitly, check if it's zero and throw an error. This will allow the adaptive algorithms to automatically set it but also have it explicit in this code.

Reply:
I did passed default as `0.0` but it still gives the same error.

Reply:
no, take it as input. Do what you were doing before, but with a default value to `dt` of zero.

Reply:
Oh.. ok.